sonic-net/sonic-mgmt
# feat: add BFD traffic test

### Description of PR

Implement BFD PTF-based traffic test.

Summary:
Fixes # (issue) Microsoft ADO 28382949

### Type of change





### Back port request

### Approach
#### What is the motivation for this PR?
We need to confirm the patch switching behavior when one of the backend port channels is down in a T2 chassis.

#### How did you do it?
Test case 1: remote port channel shutdown
1. send 10k pkt from src PTF to dst's BGP neighbor to find the remote (dst) backend portchannel in use
2. shut down the remote backend portchannel found in step 1
3. traffic now should shift to an alternative backend portchannel
4. send 10k pkt again to verify the new traffic path is in use now

Test case 2: local port channel shutdown
1. send 10k pkt from src PTF to dst's BGP neighbor to find the local (src) backend portchannel in use
2. shut down the local backend portchannel found in step 1
3. traffic now should shift to an alternative backend portchannel
4. send 10k pkt again to verify the new traffic path is in use now

Test case 3: remote port channel member shutdown
1. send 10k pkt from src PTF to dst's BGP neighbor to find the remote (dst) backend portchannel member in use
2. shut down the remote backend portchannel member found in step 1
3. traffic now should shift to an alternative backend portchannel member
4. send 10k pkt again to verify the new traffic path is in use now

Test case 2: local port channel member shutdown
1. send 10k pkt from src PTF to dst's BGP neighbor to find the local (src) backend portchannel member in use
2. shut down the local backend portchannel member found in step 1
3. traffic now should shift to an alternative backend portchannel member
4. send 10k pkt again to verify the new traffic path is in use now

#### How did you verify/test it?
I ran the traffic test cases I added in a T2 testbed and I can see they passed

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?
This test only supports T2 topology

### Documentation
